internal/pkg/service/objectstorage/cloud/aws/s3.go

```diff
@@ -65,6 +65,36 @@ func (s S3ObjectStorageAdapter) CreateBucket(ctx context.Context, bucket *v1alph
 }
 
 func (s S3ObjectStorageAdapter) DeleteBucket(ctx context.Context, bucket *v1alpha1.Bucket) error {
+	// First we need to empty the bucket
+	in := &s3.ListObjectsV2Input{Bucket: &bucket.Spec.Name}
+	for {
```

Comment:
Can you please use a while instead ?

Maybe something like `while out, err := s.s3Client.ListObjectsV2(ctx, in); err != nil`?

Reply:
Actually, maybe you want to use this instead https://stackoverflow.com/questions/46795963/how-to-empty-a-s3-bucket-using-the-aws-sdk-for-golang

Reply:
Oh true :D But i would still have a condition in there :D

Reply:
I already saw that method with `s3manager` but it is implemented in the `aws-sdk-go` and we use `aws-sdk-go-v2`.
So I found another method to do it with the `aws-sdk-go-v2` sdk...

Reply:
Sure but I think this https://github.com/aws/aws-sdk-go-v2/issues/1463#issue-1030827057 is better :D